src/main/scala/com/nec/cache/TransferDescriptor.scala

```diff
@@ -22,3 +23,47 @@ trait TransferDescriptor {
     array.toSeq
   }
 }
+
+object TransferDescriptor {
+  def vectorAlignedSize(size: Long): Long = {
+    val dangling = size % 8
+    if (dangling > 0) {
+      // If the size is not aligned on 8 bytes, add some padding
+      size + (8 - dangling)
+    } else {
+      size
+    }
+  }
+
+  def bufferSize(columns: Seq[UnitColVector]): Long = {
+    require(columns.nonEmpty, "Need more than 1 columns to compute buffer size!")
```

Comment:
This message looks a bit wrong, did you mean to say "need more than 0" or "Need at least 1"?

Reply:
Ah yes, that's a misspelling